lib/OnyxUtils.ts

```diff
@@ -1417,4 +1419,50 @@ const OnyxUtils = {
     getEvictionBlocklist,
 };
 
+GlobalSettings.addGlobalSettingsChangeListener(({enablePerformanceMetrics}) => {
+    if (!enablePerformanceMetrics) {
+        return;
+    }
+    // We are reassigning the functions directly so that internal function calls are also decorated
+
+    // @ts-expect-error Reassign
+    initStoreValues = decorateWithMetrics(initStoreValues, 'OnyxUtils.initStoreValues');
+    // @ts-expect-error Reassign
+    maybeFlushBatchUpdates = decorateWithMetrics(maybeFlushBatchUpdates, 'OnyxUtils.maybeFlushBatchUpdates');
+    // @ts-expect-error Reassign
+    batchUpdates = decorateWithMetrics(batchUpdates, 'OnyxUtils.batchUpdates');
+    // @ts-expect-error Complex type signature
+    get = decorateWithMetrics(get, 'OnyxUtils.get');
+    // @ts-expect-error Reassign
+    getAllKeys = decorateWithMetrics(getAllKeys, 'OnyxUtils.getAllKeys');
+    // @ts-expect-error Reassign
+    getCollectionKeys = decorateWithMetrics(getCollectionKeys, 'OnyxUtils.getCollectionKeys');
+    // @ts-expect-error Reassign
+    addAllSafeEvictionKeysToRecentlyAccessedList = decorateWithMetrics(addAllSafeEvictionKeysToRecentlyAccessedList, 'OnyxUtils.addAllSafeEvictionKeysToRecentlyAccessedList');
+    // @ts-expect-error Reassign
+    keysChanged = decorateWithMetrics(keysChanged, 'OnyxUtils.keysChanged');
```

Comment:
❤️ love this!!

Reply:
yeah i only hate the `ts-expect-error` suppressions 🤔